discordbot/merriam-webster/client.go

```diff
@@ -24,26 +24,33 @@ func init() {
 // Client describes access to the Merriam-Webster dictionary at dictionaryapi.com
 type Client interface {
 	SearchCollegiate(word string) ([]types.CollegiateResult, []string, error)
+	IsConfigredCorrectly() bool
 }
 
 // BasicClient is a basic HTTP-based client for querying the M-W dictionary
 type BasicClient struct {
-	APIKey    string
-	BaseURL   *url.URL
-	UserAgent string
-	Client    *http.Client
+	APIKey          string
+	BaseURL         *url.URL
+	UserAgent       string
+	Client          *http.Client
+	InitFailedError error
 }
 
 // NewBasicClient creates a client based on a given API key
-func NewBasicClient(apiKey string, userAgent string) *BasicClient {
+func NewBasicClient(apiKey string, userAgent string, err error) *BasicClient {
```

Comment:
As `InitFailedError` lives in NewBasicClient, and is specific to client internals, it should probably live here, not in `ProvideBasicClient`. The Wire provider code should be kept as simple as possible; Wire provides enough indirection headaches already.

```go
var err error
if apiKey == "" {
  err = errors.New("Missing API key")
}
```